.github/workflows/checkpatch.yml

```
@@ -10,16 +10,19 @@ jobs:
       BUILD_DIR: ../build
     steps:
       - name: Checkout Code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
         with:
-            fetch-depth: 50
+            fetch-depth: 0
+      - name: Checkout Base
+        run: |
+            git fetch origin ${{ github.event.pull_request.base.ref }}
```

If you'd like, the FETCH_HEAD revision can be dumped to the log - to make future debugging easier.

```suggestion
            git fetch origin ${{ github.event.pull_request.base.ref }}
            echo "The current base for checkpatch is: $(git show FETCH_HEAD --oneline --raw)"            
```